core/src/com/unciv/models/ruleset/Ruleset.kt

```diff
@@ -604,7 +611,7 @@ class Ruleset {
             for (building in era.settlerBuildings)
                 if (building !in buildings)
                     lines += "Nonexistent building $building built by settlers when starting in ${era.name}"
-            if (era.startingMilitaryUnit !in units)
+            if (era.startingMilitaryUnitCount != 0 && era.startingMilitaryUnit !in units)
```

Comment:
I don't think this catches all uses, as `Difficulties.json` can also use the Era Starting Unit